### PR TITLE
Add Multi Upload Functionality

### PR DESCRIPTION
**Pull request by spaulmark**
Now you can upload a bunch of images, and they get automagically placed into characters, the names are prepopulated by the filenames.

### CODE REVIEW

File: hunger_games_simulator.ts [hunk 1]
```diff
@@ -516,6 +542,26 @@ const UI = {
...
+        OpenMultiFileDialog({
+            title: 'Bulk Add Images',
+            title_style: {
+                background: light_accent,
+                color: '#463c2a'
+            } as CSSStyleDeclaration,
+            preserve_extern_urls: true,
+            description: 'Click below to upload multiple images. This will create tributes with names identical to the filenames of each image uploaded.',
+        }).then( (dialog: DialogData) => {
+            if (!dialog.file_list) return;
```
Comment (Sirraide) [inline hunger_games_simulator.ts]: This throws if the user clicks ‘Cancel’ since `dialog` is `null` in that case.

File: js/dialog.ts [hunk 2]
```diff
@@ -181,6 +186,38 @@ export class Dialog {
...
+            event.preventDefault();
+            input.click();
+        }
+
+        let _this = this
+        input.onchange = function (event) {
+            if (!event.target) return
+            _this.data.file_list = (<HTMLInputElement>event.target)?.files
+            if (_this.data.file_list) {
+                _this.accept()
```
Comment (Sirraide) [inline js/dialog.ts]: I'd prefer it if, similarly to how the ‘Select an image’ dialog works, selected images were first displayed in the dialog (their filenames that is), and only if the user then pressed ‘Ok’ would they then actually get added as characters.

**Comment by Sirraide** [comment]
Looking good. The rest is fine I'd say.

**Comment by spaulmark** [comment]
Made the requested changes

**Pull request merged**